plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/util/FeatureDevClientUtil.kt

```diff
@@ -28,189 +29,215 @@ import software.aws.toolkits.jetbrains.services.amazonqFeatureDev.clients.Genera
+                    taskAssistConversationResult.responseMetadata().requestId()
+                }"
+            }
 
-            if (e is software.amazon.awssdk.services.codewhispererruntime.model.ServiceQuotaExceededException) {
-                throw MonthlyConversationLimitError(errMssg, e.cause)
+            return conversationId
+        } catch (e: Exception) {
+            logger.warn(e) { "$FEATURE_NAME: Failed to start conversation: ${e.message}" }
+            failureReason = e.javaClass.simpleName
+            result = Result.Failed
+            var errMssg = e.message
+            if (e is CodeWhispererRuntimeException) {
+                errMssg = e.awsErrorDetails().errorMessage()
+                logger.warn(e) { "Start conversation failed for request: ${e.requestId()}" }
+
+                if (e is software.amazon.awssdk.services.codewhispererruntime.model.ServiceQuotaExceededException) {
+                    throw MonthlyConversationLimitError(errMssg, e.cause)
+                }
             }
+            apiError(errMssg, e.cause)
+        } finally {
+            AmazonqTelemetry.startConversationInvoke(
+                amazonqConversationId = conversationId,
+                result = result,
+                reason = failureReason,
+                duration = (System.currentTimeMillis() - startTime).toDouble(),
+                credentialStartUrl = getStartUrl(project = this.project),
+                requestServiceType = "amazoncodewhispererservice"
+            )
         }
-        apiError(errMssg, e.cause)
     }
-}
-
-fun createUploadUrl(proxyClient: FeatureDevClient, conversationId: String, contentChecksumSha256: String, contentLength: Long):
-    CreateUploadUrlResponse {
-    try {
-        logger.debug { "Executing createUploadUrl with conversationId $conversationId" }
-        val uploadUrlResponse = proxyClient.createTaskAssistUploadUrl(
-            conversationId,
-            contentChecksumSha256,
-            contentLength
-        )
-        logger.debug {
-            "$FEATURE_NAME: Created upload url: {uploadId: ${uploadUrlResponse.uploadId()}, requestId: ${uploadUrlResponse.responseMetadata().requestId()}}"
-        }
-        return uploadUrlResponse
-    } catch (e: Exception) {
-        logger.warn(e) { "$FEATURE_NAME: Failed to generate presigned url: ${e.message}" }
 
-        var errMssg = e.message
-        if (e is CodeWhispererRuntimeException) {
-            errMssg = e.awsErrorDetails().errorMessage()
-            logger.warn(e) { "Create UploadUrl failed for request: ${e.requestId()}" }
-
-            if (e is ValidationException && e.message?.contains("Invalid contentLength") == true) {
-                throw ContentLengthError(message("amazonqFeatureDev.content_length.error_text"), e.cause)
+    fun createUploadUrl(conversationId: String, contentChecksumSha256: String, contentLength: Long):
```

Comment:
Do we not need to add telemetry for `createUploadUrl` API ?

Reply:
It is added in a different place, right? And it captures duration/errors for the zipping part too.